core/controllers/reader.py

```diff
@@ -300,7 +300,121 @@ def get(self, exploration_id):
+    """
+
+    def __init__(self, *args, **kwargs):
+        """Method that initializes member variables for the handler.
+
+        Attributes:
+            exp_issues: ExplorationIssues. The exploration issues domain object.
+            issue_schema_version: int. The issue schema version.
+        """
+        super(StorePlaythroughHandler, self).__init__(*args, **kwargs)
+        self.exp_issues = None
+        self.issue_schema_version = None
+
+    def _find_playthrough_in_exp_issues(self, playthrough):
+        """Finds the given playthrough in the unresolved issues list of the
+        exploration issues model, returns None if it doesn't exist.
+
+        Args:
+            playthrough: Playthrough. The playthrough domain object.
+
+        Returns:
+            int|None. The index at which the issue was found, None otherwise.
+        """
+        for index, issue in enumerate(self.exp_issues.unresolved_issues):
+            if issue.issue_type == playthrough.issue_type:
+                issue_customization_args = issue.issue_customization_args
+                # In case issue_keyname is 'state_names', the ordering of the
+                # list is important i.e. [a,b,c] is different from [b,c,a].
+                issue_keyname = stats_models.ISSUE_TYPE_KEYNAME_MAPPING[
+                    issue.issue_type]
+                if (issue_customization_args[issue_keyname] ==
+                        playthrough.issue_customization_args[issue_keyname]):
+                    return index
+        return None
+
+    def _move_playthrough_to_correct_issue(self, playthrough, orig_playthrough):
+        """Moves the updated playthrough to its correct issue in the unresolved
+        issues list.
+
+        Args:
+            playthrough: Playthrough. The updated playthrough domain object.
+            orig_playthrough: Playthrough. The original playthrough domain
+                object.
+        """
+        max_playthroughs_per_issue = False
+        issue_index = self._find_playthrough_in_exp_issues(playthrough)
+        # Check whether the playthrough can be added to its new issue,
+        # if not, it stays in its old issue.
+        if issue_index is not None:
+            if (len(self.exp_issues.unresolved_issues[
+                    issue_index].playthrough_ids) < (
+                        feconf.MAX_PLAYTHROUGHS_FOR_ISSUE)):
+                self.exp_issues.unresolved_issues[
+                    issue_index].playthrough_ids.append(playthrough.id)
+            else:
+                max_playthroughs_per_issue = True
+        else:
+            issue = stats_domain.ExplorationIssue(
+                playthrough.issue_type,
+                playthrough.issue_customization_args,
+                [playthrough.id], self.issue_schema_version, is_valid=True)
+            self.exp_issues.unresolved_issues.append(issue)
+
+        # Now, remove the playthrough from its old issue.
+        if not max_playthroughs_per_issue:
```

Comment:
So, we're only removing it if we *haven't* hit the max playthroughs?

Reply:
@brianrodri Yes. If the target issue we need to move the playthrough to already has the max playthroughs, then we don't change the issue at all.
For instance , we are supposed to change a playthrough from 'EarlyQuit at card1' to 'MultipleIncorrectSubmissions at card2', but the second issue already has maximum playthroughs, then we keep the playthrough in the first issue.

Reply:
Aha, understood. Gotcha!